CommonCPPCTest.cmake

```diff
@@ -47,6 +48,13 @@ list(SORT TEST_FILES)
 set(ALL_CPP_TESTS)
 set(ALL_CPP_PERF_TESTS)
 
+set(TEST_ENABLE_BOOST_HEADER "Add -DBOOST_TEST_DYN_LINK to tests" CACHE BOOL ON)
```

Comment:
Making it an option is a bit pointless, since in superproject builds each project needs a different setting. Why not have it a simple, non-cache variable projects set in their tests/CMakeLists.txt?